metricbeat/tests/system/test_zookeeper.py

```diff
@@ -78,6 +78,34 @@ def test_output(self):
 
         self.assert_fields_are_documented(evt)
 
+    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
+    @attr('integration')
+    def test_connections(self):
+        """
+        ZooKeeper server module outputs an event.
+        """
+        self.render_config_template(modules=[{
+            "name": "zookeeper",
+            "metricsets": ["connections"],
+            "hosts": self.get_hosts(),
+            "period": "5s"
+        }])
+        proc = self.start_beat()
+        self.wait_until(lambda: self.output_lines() > 0)
```

Comment:
Is there going to be always connections available to report?
Should we keep a connection with zookeeper open here to avoid flakiness?

Reply:
I don't follow you. If you want to change the behaviour of the module or the system tests it should go in a different PR, this is just to add a Metricset with the current behaviour and Cloud is waiting for it :wink:

Reply:
If there is no connection to the zookeeper service, no event is generated and then this will fail. I was wondering if we should open a connection in the test to avoid flakiness because of that.
Nothing to change in the module, just maybe in the test.
If there is always going to be a connection here because of other reasons then you can ignore my comment :slightly_smiling_face:

Reply:
Also this, if it is not failing now I think it won't fail, to be revisited if there are related flaky tests.